### ISSUE TITLE
Accepting without breaking the current snippet.

### ISSUE THREAD

**Issue by Cih2001** [Saghen/blink.cmp#624]
### Feature Description

I looked at the documentations. There are two commands available to accept the suggested item `accept` and `select_and_accept`. However, they both break the snippet, so if I accept a suggestion for the first parameter, then snippet is broken and I cant jump to the next part using `snippet_forward` or `snippet_backward`. Is there a way to accept a suggestion without breaking the current snippet?

![Screenshot from 2024-12-17 19-51-18](https://github.com/user-attachments/assets/5a5efce6-11cd-4a93-943f-8be0b19bf24d)

Using "super-tab" preset, here after hitting tab, I can select `testString` but cannot jump to the next parameter anymore as the snippet is broken.

**Comment by Cih2001** [comment]
Hey, Thanks for your fast response. I think I didn't provide a good enough example, and weren't probably clear with my explanations, So the suggested fix is not what exactly I wanted. So let me try again.

It seems that your fix works as long as the first parameter is just a simple string. In my example above, `testString` is actually a variable define 2 lines above where it is used. Consider the following code:

![Screenshot from 2024-12-19 23-10-43](https://github.com/user-attachments/assets/83dc2fb2-0737-4c66-a7c6-1c50304ce030)

Now if I hit tab, I expect the first parameter of the snippet is set to the variable `format` defined above, while being able to still to forward to the next parameter.

The keymap below is exactly what I want to achieve.

```
["<Tab>"] = {
	function(cmp)
		if cmp.snippet_active() then
			cmp.accept()
			return cmp.snippet_forward()
		end
		return cmp.select_and_accept()
	end,
	"fallback",
}
```

With your fix, this works as long as the first parameter is a string, not a variable or any other type of suggestion.

**Comment by Sahas-Ananth** [comment]
Is it possible you have (by default) set `completion.accept.auto_brackets.enabled = true`? I had this and it broke my snippets. Once I turned it off, functions and stuff started working without snippets breaking.

**Comment by Cih2001** [comment]
> @Cih2001 Is it possible you have (by default) set `completion.accept.auto_brackets.enabled = true`? I had this and it broke my snippets. Once I turned it off, functions and stuff started working without snippets breaking.

No, I disabled that, the problem still persist. It's not because of the auto brackets, since like I mentioned in my previous comment, it works fine if the first parameter is a string, so https://github.com/Saghen/blink.cmp/commit/284dd37f9bbc632f8281d6361e877db5b45e6ff0 partially worked.

**Comment by Cih2001** [comment]
Seems that it's fixed in one of the latest releases so I'll close it for now.
